src/components/Activities/Activity3/Game/FeaturesSlidebar.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useContext, useEffect, useState } from 'react';
+import '../../../styles/Activity3Game.scss';
+import { GameContext } from '.';
+
+interface FeatureSlidebarProps {
+  featureWeights: number[],
+  setFeatureWeights: (featureweights: number[]) => void,
+}
+
+function FeatureSlidebar(props: FeatureSlidebarProps): JSX.Element {
+  const { goNextState, variableSelection } = useContext(GameContext);
+  const { featureWeights, setFeatureWeights } = props;
+  const [weight1, setWeight1] = useState(featureWeights[0]);
+  const [weight2, setWeight2] = useState(featureWeights[1]);
+
+  useEffect(() => {
+    const newWeights = [weight1, weight2 - weight1, 100 - weight2];
+    setFeatureWeights(newWeights);
+  }, [weight1, weight2]);
```

Comment:
should apply these changes and the code changes below
```suggestion
  const [weights, setWeights] = useState(featureWeights);

  useEffect(() => {
    setFeatureWeights(weights);
  }, [weights]);

  // keeps 1st + 2nd elemnt same, sets 3rd as 100 - 1st - 2nd
  // index will only ever be 1 or 2
  const setWeight = (index: number, value: number) => {
    let tempFeatureWeights = [value, weights[1], 100 - value - weights[1]];
    if (index === 1){
      tempFeatureWeights = [weights[0], value, 100 - weights[0] - value];
    }
  }
```

Reply:
whats the point of the setWeight function? Like it doesn't set the state or return anything

Reply:
yess please add a `setWeights` state if you do want to use this! also feel free to not as well :))

Reply:
I changed the state to just be an array. I didn't use the setWeight function.